dallinger/command_line/docker_ssh.py

```diff
@@ -587,7 +587,7 @@ def get_docker_compose_yml(
     postgresql_password: str,
 ) -> str:
     """Generate a docker-compose.yml file based on the given"""
-    docker_volumes = config.get("docker_volumes", "[]")
+    docker_volumes = config.get("docker_volumes", "")
```

Comment:
Having thought about it some more, I think this is the only change that needs to happen. I think your new logic 

```
if docker_volumes | length > 0
```

should then be redundant, and can be reverted?

Reply:
Changing that one line is exactly what I did first. Unfortunately, it resulted in a subsequent error:
```
1 error(s) decoding:

* error decoding 'volumes[1]': invalid empty volume spec

`docker compose` logs failed to run.
Traceback (most recent call last):
  File "/home/frank/.virtualenvs/dallinger/bin/dallinger", line 33, in <module>
    sys.exit(load_entry_point('dallinger', 'console_scripts', 'dallinger')())
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/frank/.virtualenvs/dallinger/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/frank/projects/Dallinger/dallinger/command_line/docker_ssh.py", line 237, in wrapper
    return f(image_name, *args, **kwargs)
  File "/home/frank/projects/Dallinger/dallinger/command_line/docker_ssh.py", line 366, in deploy
    executor.run(
  File "/home/frank/projects/Dallinger/dallinger/command_line/docker_ssh.py", line 653, in run
    raise ExecuteException(
dallinger.command_line.docker_ssh.ExecuteException: An error occurred when running the following command on the remote server: 
docker compose -f ~/dallinger/bartlett-x/docker-compose.yml run --rm web ls
```

I supposed above is caused by entering the for loop with an array that contains an empty string as sole element. To illustrate:
```
>>> docker_volumes = ""
>>> docker_volumes.split(",")
['']
```
This would then result in an `invalid empty volume spec` error.

Reply:
I see, sorry for my bad advice here!